pkg/utils/util.go

```diff
@@ -348,15 +348,45 @@ func GetDefaultPipelineBundleRef(buildPipelineSelectorYamlURL, selectorName stri
 	if err = yaml.Unmarshal(body, ps); err != nil {
 		return "", fmt.Errorf("failed to unmarshal build pipeline selector: %v", err)
 	}
-	for _, s := range ps.Spec.Selectors {
+	for i := range ps.Spec.Selectors {
+		s := &ps.Spec.Selectors[i]
 		if s.Name == selectorName {
-			return s.PipelineRef.Bundle, nil //nolint:all
+			return GetBundleRef(&s.PipelineRef), nil
 		}
 	}
 
 	return "", fmt.Errorf("could not find %s pipeline bundle in build pipeline selector fetched from %s", selectorName, buildPipelineSelectorYamlURL)
 }
 
+// GetPipelineNameAndBundleRef returns the pipeline name and bundle reference from a pipelineRef
+// https://tekton.dev/docs/pipelines/pipelineruns/#tekton-bundles
+func GetPipelineNameAndBundleRef(pipelineRef *v1beta1.PipelineRef) (string, string) {
+	var name string
+	var bundleRef string
+
+	// Support the v1beta1 style
+	name = pipelineRef.Name
+	bundleRef = pipelineRef.Bundle //nolint:all
+
+	// Prefer the v1 style
```

Comment:
I think we shouldn't allow taking one value from `v1beta1` and another one from `v1`

Reply:
Changed it take the values only from `params` when `resolver` is set, otherwise only from `name`+`bundle`

(no custom error in case both are set)